src/DeviceInterface/AndroidInterface.js

```diff
@@ -11,21 +17,29 @@ class AndroidInterface extends InterfacePrototype {
      * @returns {Promise<string|undefined>}
      */
     async getAlias () {
-        const { alias } = await sendAndWaitForAnswer(async () => {
-            if (this.inContextSignup.isAvailable()) {
-                const { isSignedIn } = await this.deviceApi.request(new ShowInContextEmailProtectionSignupPromptCall(null))
+        // If in-context signup is available, do that first
+        if (this.inContextSignup.isAvailable()) {
+            const { isSignedIn } = await this.deviceApi.request(new ShowInContextEmailProtectionSignupPromptCall(null))
+            if (isSignedIn) {
                 // On Android we can't get the input type data again without
                 // refreshing the page, so instead we can mutate it now that we
                 // know the user has Email Protection available.
-                if (this.globalConfig.availableInputTypes) {
-                    this.globalConfig.availableInputTypes.email = isSignedIn
+                if (this.settings.availableInputTypes) {
+                    this.settings.setAvailableInputTypes({email: isSignedIn})
                 }
                 this.updateForStateChange()
                 this.onFinishedAutofill()
             }
-            return window.EmailInterface.showTooltip()
-        }, 'getAliasResponse')
-        return alias
+        }
+        // Then, if successful actually prompt to fill
+        if (this.settings.availableInputTypes.email) {
+            const {alias} = await this.deviceApi.request(new EmailProtectionGetAliasCall({
+                requiresUserPermission: !this.globalConfig.isApp,
+                shouldConsumeAliasIfProvided: !this.globalConfig.isApp,
+                isIncontextSignupAvailable: this.inContextSignup.isAvailable()
+            }))
+            return alias ? formatDuckAddress(alias) : alias
```

Comment:
is the idea here that we either format if `alias` is there, or return `undefined` if it isn't?

Can the call ever succeed without a value? if not, can the types/schemas better reflect it?

or, if this entire method should be able to return `Promise<undefined>` (from the signature) then could we make it more explicit? having the `alias` on both sides of the ternary was a little confusing to understand, :)

Reply:
Lol, I copy/pasted what was there previously without giving it any thought. Let me look into that tomorrow.

Reply:
I ended up returning `undefined` explicitly to avoid changing anything on the API level at this stage.